lib/src/inline_parser.dart

```diff
@@ -938,26 +958,50 @@ class TagSyntax extends InlineSyntax {
   /// [getChildren], in which [parser] parses any nested text into child nodes.
   /// The returned [Node] incorpororates these child nodes.
   Node? close(InlineParser parser, Delimiter opener, Delimiter closer,
-      {required List<Node> Function() getChildren}) {
-    var strong = opener.length >= 2 && closer.length >= 2;
-    return Element(strong ? 'strong' : 'em', getChildren());
+      {required String tag, required List<Node> Function() getChildren}) {
+    return Element(tag, getChildren());
   }
 }
 
+class EmphasisSyntax extends DelimiterSyntax {
+  /// Parses `__strong__` and `_emphasis_`.
+  EmphasisSyntax.underscore()
+      : super('_+', requiresDelimiterRun: true, tags: _tags);
+
+  /// Parses `**strong**` and `*emphasis*`.
+  EmphasisSyntax.asterisk()
+      : super(r'\*+',
+            requiresDelimiterRun: true, allowIntraWord: true, tags: _tags);
+
+  static final _tags = [DelimiterTag('em', 1), DelimiterTag('strong', 2)];
+}
+
 /// Matches strikethrough syntax according to the GFM spec.
-class StrikethroughSyntax extends TagSyntax {
+class StrikethroughSyntax extends DelimiterSyntax {
   StrikethroughSyntax()
-      : super('~+', requiresDelimiterRun: true, allowIntraWord: true);
+      : super('~+',
+            requiresDelimiterRun: true,
+            allowIntraWord: true,
+            tags: [DelimiterTag('del', 2)]);
+}
 
-  @override
-  Node close(InlineParser parser, Delimiter opener, Delimiter closer,
-      {required List<Node> Function() getChildren}) {
-    return Element('del', getChildren());
-  }
+/// Parses `==mark==` to `<mark>mark</mark>`.
```

Comment:
Can you add some documentation about where this is coming from? We need specifications which we can follow in order to support special syntaxes.

Reply:
I did not find a common used specification about it.

My idea is all the syntaxes extend `DelimiterSyntax`  with `requiresDelimiterRun = true` follow the commonmark specification of **Emphasis and strong emphasis**: https://spec.commonmark.org/0.30/#emphasis-and-strong-emphasis

Reply:
I don't think we can support an unspecified, or loosely specified syntax. Even if it is not a complicated one, it adds maintenance burden, and can create breaking changes when a well-specified, popular syntax _does_ emerge, and is slightly different from what we've implemented. As is noted in the issue, some implementations seem to support different delimiters for highlighting, so I think this is too under-specified to accept.

I'll comment more on the issue.